### DISCUSSION TITLE
Get collision event forces?

### DISCUSSION THREAD

**Discussion by tversteeg** [jcornaz/heron#187] [Q&A]
Is it possible to get any extra information when a collision occurs, such as the impact force? When I print an event using `EventReader<CollisionEvent>` I just get the following information:

```rust
event = Started(
    CollisionData {
        rigid_body_entity: 8v0,
        collision_shape_entity: 8v0,
        collision_layers: CollisionLayers {
            groups: 4294967295,
            masks: 4294967295,
        },
    },
    CollisionData {
        rigid_body_entity: 19v0,
        collision_shape_entity: 19v0,
        collision_layers: CollisionLayers {
            groups: 4294967295,
            masks: 4294967295,
        },
    },
)
```

I would like to see with how much force the two objects collided. I tried looking into the attached collider entities, but these don't have this kind of information.

**Comment by jcornaz** [comment]
Mmh, interesting requirement.

I don't see how to provide such information. The event we get from rapier only contains a reference to the collider (which in turns has a reference to the rigid body): https://docs.rs/rapier3d/latest/rapier3d/geometry/enum.ContactEvent.html

And I don't think we can get information about the collision from either the [collider](https://docs.rs/rapier3d/latest/rapier3d/geometry/struct.Collider.html) or [rigid body](https://docs.rs/rapier3d/0.11.1/rapier3d/dynamics/struct.RigidBody.html).

**Reply by tversteeg** [reply]
Thanks for the reply. For now I'm using the velocity of the object since most collisions are against static objects, but it's not really ideal.

**Reply by jcornaz** [reply]
Yes of course, this isn't ideal.

After searching more, we may be able to extract the [ContactData](https://docs.rs/rapier3d/latest/rapier3d/geometry/struct.ContactData.html) from rapier's [contact graph](https://www.rapier.rs/docs/user_guides/rust/advanced_collision_detection#the-contact-graph). Which contains the impulse info.
